pennylane/workflow/execution.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the :class:`ExecutionConfig` data class.
+"""
+from dataclasses import dataclass
+from typing import Optional, Tuple, Union
+
+from pennylane.interfaces import SUPPORTED_INTERFACES
+from pennylane.gradients import SUPPORTED_GRADIENT_KWARGS
+
+SUPPORTED_GRADIENT_METHODS = [
+    "best",
+    "parameter-shift",
+    "backprop",
+    "finite-diff",
```

Comment:
We will need to add spsa after https://github.com/PennyLaneAI/pennylane/pull/3440

Reply:
I'd like for this list to ultimately exist in only one place after this overhaul